### PR TITLE
Split out running terraform apply and notifying proton to separate jobs

### PR DESCRIPTION
**Pull request by awishn02**
*Description of changes:* Basically this just removes the need to have "continue-on-error" all over the place. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tf_tutorial_env/.proton/deployment-metadata.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (augentry) [inline tf_tutorial_env/.proton/deployment-metadata.json:1]: looks like you included your env accidentally?

File: .github/workflows/proton_run.yml [hunk 2]
```diff
@@ -160,45 +164,76 @@ jobs:
...
+
+  notify-proton:
+    name: 'Notify Proton'
+    needs: 
+    - get-deployment-data
+    - terraform
+    runs-on: ubuntu-latest
+    environment: ${{ needs.get-deployment-data.outputs.environment }}
+
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```
Comment (augentry) [inline .github/workflows/proton_run.yml:198]: Do we want to run ANY of this workflow when we aren't merging to main? Seems like things could get out of sync if we run terraform apply on a branch. I was doing the following and I thought it was the most specific:
```
  push:
    branches:
      - main
    paths:
      - '**/.proton/deployment-metadata.json'
```

File: tf_tutorial_env/.proton/deployment-metadata.json (hunk 1 above)
Comment (awishn02) [reply tf_tutorial_env/.proton/deployment-metadata.json]: yep, removed

File: .github/workflows/proton_run.yml (hunk 2 above)
Comment (awishn02) [reply .github/workflows/proton_run.yml:198]: yeah i actually have that same config, so this main check is not necessary, no harm though also

**Review by augentry — APPROVED** [review]

**Pull request merged**